docs/tutorials/domain_tutorial.jl

```diff
@@ -0,0 +1,140 @@
+Depending on the domain, the returned coordinate field will have elements of different names and types. For example,
+the SphericalShell domain has coordinates of latitude, longitude, and height, while a Plane domain has coordinates
+of x and y, and a Point domain only has a coordinate z_sfc.
+
+
+## Domain types
+All ClimaLSM domains are subtypes of abstract type `ClimaLSM.Domains.AbstractDomain`.
+A variety of concrete domain types are supported:
+
+- 0D: `Domains.Point`
+- 1D: `Domains.Column`
+- 2D: `Domains.Plane`, `Domains.SphericalSurface`
+- 3D: `Domains.HybridBox`, `Domains.SphericalShell`.
+
+Single component models (soil, snow, vegetation, canopy airspace...) will be solved on 
+a single domain. Which domain is appropriate depends on the model equations (PDE vs ODE) and
+on the configuration of interest (single column or 3d). 
+
+Furthermore, these individual model domains are composed when building a multi-component
+Land System Model. In this case, the full LSM domain is split into regions:
+the subsurface and the surface[^2]. Different land model components are associated
+with these different regions - e.g. the soil model is a subsurface model, while vegetation
+is a surface model. The above domain types are then paired based on the desired LSM
+domain being modeled:
+
+| LSM Domain | Surface Domain | Subsurface Domain |
+| :---         |     :---:      |          ---: |
+| LSMSingleColumnDomain   | Point    | Column    |
+| LSMCartesianBoxDomain    | Plane[^3]       | HybridBox      |
+| LSMSphericalShellDomain    | SphericalSurface[^3][^4]       | SphericalShell      |
+
+It is important to note that the horizontal domain used for the surface and subsurface 
+domains are identical in LSM simulations. This ensures that we can use the same indexing
+of surface and subsurface domains and variables to correctly compute and 
+apply boundary fluxes between different component models. Otherwise we would need
+to develop additional infrastructure in order to, for example, select the correct subsurface
+column corresponding to a particular surface location.
+
+
+
+## How variable initialization depends on domains
+When a developer creates a model, they need to specify the symbols used for the prognostic variables,
+via [`prognostic_vars`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.prognostic_vars),
+and
+the types of those variables,
+ via [`prognostic_types`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.prognostic_types).
+
+The [`initialize`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.initialize)
+function (which calls both
+[`initialize_prognostic`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.initialize_prognostic)
+ and [`initialize_auxiliary`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.initialize_auxiliary)
```

Comment:
Missing closing bracket
```suggestion
via [`prognostic_vars`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.prognostic_vars),
and
the types of those variables,
 via [`prognostic_types`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.prognostic_types).

The [`initialize`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.initialize)
function (which calls both
[`initialize_prognostic`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.initialize_prognostic)
 and [`initialize_auxiliary`](https://clima.github.io/ClimaLSM.jl/dev/APIs/SharedUtilities/#ClimaLSM.initialize_auxiliary))
```